### DISCUSSION TITLE
Why the `strikethrough` mark can't be rendered?

### DISCUSSION THREAD

**Discussion by 2dogchan** [neovim/neovim#24346] [Q&A]
> 2023/07/19
> This problem just occurs on `macos`.

I tried to print a strikethrough in the built-in terminal of nvim with `wezterm`, it works unexpectlly. But it works fine on `neovide`.

The effect in `Vim 9.0`:
![image](https://github.com/neovim/neovim/assets/94974214/359b6256-3e17-4d0b-bf0b-ec356d57de4c)
And in `Nvim nightly`
![image](https://github.com/neovim/neovim/assets/94974214/3c099a6c-dc72-43a4-a13c-ffa978222f01)

The following code is the `Terminal Info`
```
--- Terminal info --- {{{
&term: xterm-256color
Description: xterm with 256 colors
Aliases: xterm-256color
Boolean capabilities:
auto_left_margin          bw         = false
auto_right_margin         am         = true
no_esc_ctlc               xsb        = false
ceol_standout_glitch      xhp        = false
eat_newline_glitch        xenl       = true
erase_overstrike          eo         = false
generic_type              gn         = false
hard_copy                 hc         = false
has_meta_key              km         = true
has_status_line           hs         = false
insert_null_glitch        in         = false
memory_above              da         = false
memory_below              db         = false
move_insert_mode          mir        = true
move_standout_mode        msgr       = true
over_strike               os         = false
status_line_esc_ok        eslok      = false
dest_tabs_magic_smso      xt         = false
tilde_glitch              hz         = false
transparent_underline     ul         = false
xon_xoff                  xon        = false
needs_xon_xoff            nxon       = false
prtr_silent               mc5i       = true
hard_cursor               chts       = false
non_rev_rmcup             nrrmc      = false
no_pad_char               npc        = true
non_dest_scroll_region    ndscr      = false
can_change                ccc        = true
back_color_erase          bce        = true
hue_lightness_saturation  hls        = false
col_addr_glitch           xhpa       = false
cr_cancels_micro_mode     crxm       = false
has_print_wheel           daisy      = false
row_addr_glitch           xvpa       = false
semi_auto_right_margin    sam        = false
cpi_changes_res           cpix       = false
lpi_changes_res           lpix       = false
backspaces_with_bs        OTbs       = true
crt_no_scrolling          OTns       = false
no_correctly_working_cr   OTnc       = false
gnu_has_meta_key          OTMT       = false
linefeed_is_newline       OTNL       = false
has_hardware_tabs         OTpt       = false
return_does_clr_eol       OTxr       = false
```

My nvim version:
```bash
NVIM v0.10.0-dev-669+g9176b5e10
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
```

**Comment by clason** [comment]
Depends on the terminal. Some support it, some don't.

**Reply by 2dogchan** [reply]
Pay attention to the picture, they're works on the same termial `wezterm`, that's why I'm confusing about it.

**Reply by clason** [reply]
Sorry, screenshots are much less useful than a text description (and this one is very hard to read).

**Reply by 2dogchan** [reply]
That's my mistake, I'll describe it in more detail.

**Comment by zeertzjq** [comment]
Try checking the value of the 'termguicolors' option. This seems to happen if the program inside the terminal is using RGB colors, but 'termguicolors' is not enabled in Nvim.

**Reply by 2dogchan** [reply]
Shouldn't be that, the option's value had setted to `true`.

**Comment by thingsiplay** [comment]
I just opened up NeoVim and it's internal terminal with `:term`, and it worked. At least on Linux. Notice this is Nvim 0.9. Maybe something changed in 0.10?

![2023-08-06_19-19](https://github.com/neovim/neovim/assets/1114291/de24669b-2f5e-4969-8a2a-3240989a7cc2)

**Comment by f3nrir92** [comment]
@2dogchan Hello! Does you have any progress here? Got same issue

**Reply by 2dogchan** [reply]
Sorry, I still have no way to solve it.

**Reply by 2dogchan** [reply]
Actually, I try to render the strikethrough on the pure nvim( without any plugins ), but also can't be work.

**Comment by sibevin** [comment]
I encountered the same issue, but eventually discovered that the colorscheme plugin was outdated. After updating it, it displayed correctly.

**Reply by f3nrir92** [reply]
Unfortunately nether changing/updating or using default color scheme does nothing for me. 
@sibevin could you tell please what exactly did you update?

![image](https://github.com/neovim/neovim/assets/5359356/148a533f-0c15-455c-a0f3-fae0f1c5091b)

**Reply by sibevin** [reply]
I use **onedarkpro.nvim** (https://github.com/olimorris/onedarkpro.nvim) as colorscheme, so I just update this plugin to latest version. My situation is that after I updated **neorg**, the strikethrough formatting is not displaying correctly. Because it was working before the update, I can confirm that it’s not a terminal or nvim issue. Therefore, I suspect it might be related to the colorscheme settings.

**Reply by f3nrir92** [reply]
Thanks. It seems that in my case some other plugin breaks proper displaying of strikethrough text. I will try to research it.

**Reply by evpeople** [reply]
#3436   I think this is helpful, I can display strikethrough by 
```
infocmp $TERM > myterm.info
vim myterm.info # add smxx=\E[9m, rmxx=\E[29m,
tic -x myterm.info

**Comment by eli-front** [comment]
> #3436 I think this is helpful, I can display strikethrough by
> 
> ```
> infocmp $TERM > myterm.info
> vim myterm.info # add smxx=\E[9m, rmxx=\E[29m,
> tic -x myterm.info
> ```

This worked perfectly. iTerm - m1 - x-term-256.
Just ran in my command line and added the "smxx=\E[9m, rmxx=\E[29m," as suggested at the bottom of myterm.info list

**Comment by kuznetsss** [comment]
I also got into the same issue and the solution is described in wezterm's docs: https://wezterm.org/config/lua/config/term.html?h=term#term-xterm-256color

Doing
```bash
$ tempfile=$(mktemp) \
  && curl -o $tempfile https://raw.githubusercontent.com/wezterm/wezterm/main/termwiz/data/wezterm.terminfo \
  && tic -x -o ~/.terminfo $tempfile \
  && rm $tempfile
```
And then adding `term = "wezterm"` to wezterm config fixed the issue.

**Comment by tcrundall** [comment]
I encountered something similar but the root cause was my `tmux` overriding `$TERM` and therefore presumably not communicating strikethrough capabilities to neovim.

If anyone else encounters this, the solution was to set `TERM` appropriately in `tmux.conf`. In my case that was
```
set -g default-terminal "xterm-ghostty"
```
You can work out what value you should be setting it to by exiting `tmux` and executing `echo $TERM`.

**Reply by wookayin** [reply]
While this would be a good workaround, having `TERM="tmux-256color"` in tmux is actually fine and would be more recommended than `"xterm-ghostty"`) --- strikethrough should work with `tmux-256color`. Whatever TERM you had before (e.g. `xterm-256color` as `default-terminal`) probably did not have the correct terminfo for strikethrough.

```bash
infocmp -1x tmux-256color | grep -E 'infocmp|smxx|rmxx'
```
which should print entires like `rmxx=\E[29m` `smxx=\E[9m` for the strikethrough capability.

**Reply by tcrundall** [reply]
Ah ok, thanks for the tip! You're right, for some reason my `tmux-256color` terminfo doesn't have `smxx` or `rmxx`. I modified the terminfo file and reverted my `default-terminal` back to `"tmux-256color"` and everything is working:

```bash
$ infocmp -1x tmux-256color | grep -E 'infocmp|smxx|rmxx'
#       Reconstructed via infocmp from file: /Applications/Ghostty.app/Contents/Resources/terminfo/74/tmux-256color
$ infocmp -1x xterm-ghostty | grep -E 'infocmp|smxx|rmxx' # for comparison
#       Reconstructed via infocmp from file: /Applications/Ghostty.app/Contents/Resources/terminfo/78/xterm-ghostty
        rmxx=\E[29m,
        smxx=\E[9m,
$ infocmp -1x tmux-256color > tmux-256color.info
$ echo "        smxx=\E[9m," >> tmux-256color.info
$ echo "        rmxx=\E[29m," >> tmux-256color.info
$ tic -x tmux-256color.info
$ infocmp -1x tmux-256color | grep -E 'infocmp|smxx|rmxx'
#       Reconstructed via infocmp from file: /Applications/Ghostty.app/Contents/Resources/terminfo/74/tmux-256color
        rmxx=\E[29m,
        smxx=\E[9m,
```
 Thanks again!
